src/Microsoft.Health.SqlServer/Features/Schema/SchemaInitializer.cs

```diff
@@ -194,31 +210,36 @@ private bool CanInitialize()
 
             if (_sqlServerDataStoreConfiguration.AllowDatabaseCreation)
             {
-                var connection = GetConnectionIfDatabaseNotExists(_sqlServerDataStoreConfiguration.ConnectionString, databaseName);
-
-                if (connection != null)
+                using (var connection = await _sqlConnectionFactory.GetSqlConnectionAsync(MasterDatabase))
```

Comment:
Is the new code actually equivalent of the previous code?

The previous code makes connection with InitialCatalog = string.Empty, this one is making connection with InitialCatalog = master?

Reply:
Yeah, this change was made after having discussion with Brandon.

Reply:
Main intention for this code is to connect to master and then check if database exists and then create database based on the result.